File: exercism/haskell/space-age/src/SpaceAge.hs
module SpaceAge (Planet(..), ageOn) where

data Planet = Mercury
            | Venus
            | Earth
            | Mars
            | Jupiter
            | Saturn
            | Uranus
            | Neptune

orbitalPeriod :: Planet -> Float
orbitalPeriod planet = case planet of
  Mercury -> 0.24084767
  Venus   -> 0.61519726
  Earth   -> 1.0
  Mars    -> 1.8808158
  Jupiter -> 11.862615
  Saturn  -> 29.447498
  Uranus  -> 84.016846
  Neptune -> 164.79132

earthYears :: Float -> Float
earthYears seconds = seconds / 31557600

ageOn :: Planet -> Float -> Float
ageOn planet seconds = (earthYears seconds) / orbitalPeriod planet

File: exercism/javascript/space-age/space-age.js
//
// This is only a SKELETON file for the 'Space Age' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const age = (planet, seconds) => {
  return Number.parseFloat((seconds / PLANETARY_YEAR_SECONDS[planet]).toFixed(2))
};

const EARTH_YEAR_SECONDS = 31557600;

const PLANETARY_YEAR_SECONDS = {
  mercury: EARTH_YEAR_SECONDS * 0.2408467,
  venus: EARTH_YEAR_SECONDS * 0.61519726,
  earth: EARTH_YEAR_SECONDS,
  mars: EARTH_YEAR_SECONDS * 1.8808158,
  jupiter: EARTH_YEAR_SECONDS * 11.862615,
  saturn: EARTH_YEAR_SECONDS * 29.447498,
  uranus: EARTH_YEAR_SECONDS * 84.016846,
  neptune: EARTH_YEAR_SECONDS * 164.79132
}
